figures/monolith-discoverers.mmd

```diff
@@ -7,6 +7,17 @@ classDiagram
     ManualMonolithDiscoverer --|> MonolithDiscoverer
     HarnessMonolithDiscoverer --|> MonolithDiscoverer
     FlyMonolithDiscoverer --|> MonolithDiscoverer
+    DNSMonolithDiscoverer --|> MonolithDiscoverer
+
+    DNSDiscoveryConfig --o DNSMonolithDiscoverer
+    DNSDiscoveryConfig : +u16 monolith_port
+    DNSDiscoveryConfig : +String dns_server
```

Comment:
Maybe this should be an `Option<String>` instead? It should default to the system configured dns server.